.github/workflows/check-code-quality.yml

```diff
@@ -55,7 +55,7 @@ jobs:
 
       - name: Install and run mypy
         run: |
-          pip install mypy
+          pip install mypy numpy
```

Comment:
@TonyXiang8787 this is a serious side-effect of the fact that we don't do `pip install -e .[dev]` in this pipeline. should we?

Reply:
maybe out of scope of this PR

Reply:
Can you elaborate what you mean?

Reply:
because `numpy` was not installed in this pipeline, the pipeline did not check mypy on any lines containing `numpy` arrays (which is quite a lot). as a result, we now did not catch stuff on mypy that we should have caught.

installing `numpy` into this pipeline enabled checking mypy on lines containing numpy arrays.

now, if, in the future, we also have other dependencies that we want to use in our own mypy checks, we would also have to add those to this pipeline.

However, if we have a step before running any of the code quality checks that does `pip install -e .[dev]`, then all those dev dependencies are already installed automatically, so we don't need to explicitly add them here.

Reply:
But you can change the order of stuff right. First install `.[dev]` then you have all the stuff.

Reply:
yes that's what I meant: do we want to do that?

Reply:
no problem

Reply:
creted #756